src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/organisations/OrganisationsResource.kt

```diff
@@ -26,15 +25,25 @@ class OrganisationsResource(
   @PutMapping("/organisations/reports/reconciliation")
   @ResponseStatus(HttpStatus.ACCEPTED)
   suspend fun generateOrganisationsReconciliationReport() {
-    val organisationsCount = nomisApiService.getCorporateOrganisationIds(0, 1).totalElements
+    val counts = reconciliationService.getOrganisationsCounts()
 
-    telemetryClient.trackEvent("organisations-reports-reconciliation-requested", mapOf("organisations" to organisationsCount.toString()))
+    telemetryClient.trackEvent("organisations-reports-reconciliation-requested", mapOf("organisations" to counts.nomisCount.toString()))
+
+    if (counts.nomisCount != counts.dpsCount) {
+      telemetryClient.trackEvent(
```

Comment:
Checks that a DPS doesn't have an orphaned organisation - this is probably good enough, but we might need to improve if this ever happens and we need to fix the one(s) that are missing.
But since there is no sync from DPS right now we can worry about that if it ever happens